pkg/util/bulk/aggregator_stats.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package bulk
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+)
+
+var tagToAggregatorEventConstructor map[string]func(aggEvent []byte) (TracingAggregatorEvent, error)
+
+// RegisterAggregatorEventConstructor registers a constructor method for a
+// particular TracingAggregatorEvent identified by its tag.
+func RegisterAggregatorEventConstructor(
```

Comment:
I think we're reinventing something gogo already has; this is basically to do same thing pbtojson does before the json step; can we just use the proto message names instead of a new notion of tag, and then the existing map from gogo?

Reply:
nice, I did not know about that map gogo maintained. I've changed `Tag` to `ProtoName` so that its clear what every `TracingAggregatorEvent` proto should implement. This allows us to then just use `protoreflect.DecodeMessage`.